src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.WebEncoders;
+
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// An <see cref="IHtmlLocalizer"/> that uses the <see cref="IStringLocalizer"/> to provide localized HTML content.
+    /// This service just encodes the arguments but not the resource string.
+    /// </summary>
+    public class HtmlLocalizer : IHtmlLocalizer
+    {
+        private IStringLocalizer _localizer;
+        private readonly IHtmlEncoder _encoder;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/> to read strings from.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizer localizer, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizer = localizer;
+            _encoder = encoder;
+        }
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key] => _localizer[key];
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key, params object[] arguments] =>
```

Comment:
Do not change at this time but remember when you touch these files again: [Engineering guidelines](https://github.com/aspnet/Home/wiki/Engineering-guidelines#argument-null-checking-in-interface-member-definitions-and-abstractvirtual-methods) say implementations of an interface that disallows `null` don't also need the `[NotNull]` attribute. I forgot that point earlier in this review, sorry.